src/main/java/frc/robot/subsystems/elevator/ElevatorIOReal.java

```diff
@@ -1,15 +1,96 @@
 package frc.robot.subsystems.elevator;
 
+import static edu.wpi.first.units.Units.Degrees;
+import static edu.wpi.first.units.Units.Meters;
+
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+import com.ctre.phoenix6.controls.StrictFollower;
+import com.ctre.phoenix6.hardware.TalonFX;
 import edu.wpi.first.units.Distance;
 import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.Servo;
+import frc.robot.Ports;
 
 public class ElevatorIOReal implements ElevatorIO {
+    private final Servo servo;
+    private final TalonFX mainMotor;
+    private final TalonFX auxMotor;
+
+    private final DigitalInput bottomSensor;
+    private final DigitalInput topSensor;
+
+    private final MotionMagicVoltage positionControl = new MotionMagicVoltage(0);
+    private final DutyCycleOut powerControl = new DutyCycleOut(0);
+
+    public ElevatorIOReal() {
+        mainMotor = new TalonFX(Ports.Elevator.MAIN_MOTOR);
+        auxMotor = new TalonFX(Ports.Elevator.AUX_MOTOR);
+        servo = new Servo(Ports.Elevator.SERVO);
+        bottomSensor = new DigitalInput(Ports.Elevator.BOTTOM_LIMIT_SWITCH);
+        topSensor = new DigitalInput(Ports.Elevator.TOP_LIMIT_SWITCH);
+
+        mainMotor.getConfigurator().apply(ElevatorConstants.MOTOR_CONFIGURATION);
+        auxMotor.getConfigurator().apply(ElevatorConstants.MOTOR_CONFIGURATION);
+
+        auxMotor.setControl(new StrictFollower(mainMotor.getDeviceID()));
+    }
+
     @Override
-    public void setPower(double power) {}
+    public void setPower(double power) {
+        mainMotor.setControl(powerControl.withOutput(power));
+    }
 
     @Override
-    public void setHeight(MutableMeasure<Distance> height) {}
+    public void setHeight(MutableMeasure<Distance> height) {
+        inputs.heightSetpoint.mut_replace(height);
+        mainMotor.setControl(positionControl.withPosition(height.in(Units.Meters)));
+    }
 
     @Override
-    public void updateInputs(ElevatorInputs inputs) {}
+    public void updateInputs(ElevatorInputs inputs) {
+        inputs.isBottom = atBottom();
+        inputs.isTop = atTop();
+        inputs.carriageHeight = Meters.of(mainMotor.getPosition().getValue()).mutableCopy();
+
+        inputs.gripperHeight.mut_replace(
+                inputs.carriageHeight.gt(ElevatorConstants.GRIPPER_HEIGHT)
```

Comment:
It's not clear what `GRIPPER_HEIGHT` means. Also, you're using both "carriage" and "gripper" -- choose one term and stick to it everywhere.

Reply:
Carriage height is the height of the Carriage and gripper height is actually the the gripper height +elevator height, you can see in the simulation video how those are different things.

Reply:
1. I didn't see any videos with the gripper. I still don't understand the difference. The gripper is the carriage (at least that's the common term I know). Perhaps you mean carriage == second stage of the elevator?
2. My point about "gripper height" is that the name is too general. Which height? Relative to what?

Reply:
1. Perhaps you mean carriage == second stage of the elevator? yes
2. the height of the connecting point between the elevator to the wrist, relatively to the floor. Is the name wristHeight better?

Reply:
1. Then I'd change everywhere the terms to how they are usually used – carriage is what the elevator carries, regardless of the amount of stages. I mean, I never saw "second carriage" in a three-stage elevator. You can clarify it with Ari or the Engineering team.
2. "gripper" and "wrist" are interchangeable so the new name won't be clearer. Why do we care about the height from the floor? Why not measure from the resting height of the gripper?

Reply:
1. Then what do you suggest?
2. Why not measure from the resting height of the gripper? My mistake, we do do it.

Reply:
1. Choose whether to use "gripper" or "carriage" and change to it – I think the first is better because it's used more widely in the code already. Where needed, use "first stage"/"second stage".
2. No probs. So is `GRIPPER_HEIGHT` needed at all?

Reply:
1. 372e293d58eecfda198c4c6ab4278d75fc9fb38f
2. We want to log both of them, cause one is relevant for the gripper and the other one for the climb.

Reply:
`GRIPPER_HEIGHT` is a constant – why log it?

Reply:
It's the height that the elevator put the gripper in.
91ff9f153b73c660fb3063263c385105f1dfaf21

Reply:
Look. Maybe, it's me, hi, I'm the problem it's me, but this is all very confusing. Again, the gripper _is_ the carriage. You should fix the terms and use first/second stage if needed.